src/wazuh.py

```diff
@@ -353,3 +354,57 @@ def _generate_cluster_snippet(
             <disabled>no</disabled>
         </cluster>
     """
+
+
+def change_api_password(username: str, old_password: str, new_password: str) -> None:
+    """Change Wazuh's API password for the default 'wazuh' user.
+
+    Args:
+        username: the username to change the user for.
+        old_password: the old API password for the user.
+        new_password: the new API password for the user.
+
+    Raises:
+        WazuhInstallationError: if an error occurs while processing the requests.
+    """
+    # The certificates might be self signed and there's no security hardening in
+    # passing them to the request since tampering with `localhost` would mean the
+    # container filesystem is compromised
+    try:
+        r = requests.get(  # nosec
+            "https://localhost:55000/security/user/authenticate",
+            auth=(username, old_password),
+            timeout=10,
+            verify=False,
+        )
+        # The old password has already been changed. Nothing to do.
+        if r.status_code == 401:
+            return
+        token = r.json()["data"]["token"]
```

Comment:
Should be defensive, the entries might not exists

Reply:
If the server replies with a 200, that's the content that should be there. If there's some unexpected condition, we'll be throwing an error and the charm will go into error status, which I think is fine. Thoughts?

Reply:
Updated